File: frontend/lib/Classes/DiscussionModel.dart
import 'package:frontend/Classes/UserB.dart';
import 'package:json_annotation/json_annotation.dart';
part 'DiscussionModel.g.dart';

@JsonSerializable()
class DiscussionModel {
  DiscussionModel(
      {required this.id,
      required this.userInfo,
      required this.payload,
      required this.messageType,
      required this.timeStamp});

  factory DiscussionModel.fromJson(Map<String, dynamic> data) =>
      _$DiscussionModelFromJson(data);
  Map<String, dynamic> toJson() => _$DiscussionModelToJson(this);

  final String id;
  final UserB userInfo;
  final String payload;
  final String messageType;
  final DateTime timeStamp;
}

File: Backend/models/DiscussionModel.js
const mongoose = require("mongoose");
const moment = require("moment");

const discussionModel = mongoose.Schema({
    userInfo : {type : mongoose.Schema.Types.ObjectId, ref : "UserB"},
    payload : String,
    messageType : String,
    timeStamp : String
});
const Model = mongoose.model("DiscussionModel", discussionModel)
module.exports = Model;